plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/AppEngineLibraryContainerResolverJob.java

```diff
@@ -58,25 +57,27 @@
   private static final Logger logger = Logger.getLogger(AppEngineLibraryContainerResolverJob.class.getName());
 
   private Map<String, Library> libraries;
-  private final IJavaProject javaProject;
   private LibraryClasspathContainerSerializer serializer;
-  private ServiceReference<ILibraryRepositoryService> serviceReference = null;
 
+  @Inject
+  private IJavaProject javaProject;
+  @Inject
   private ILibraryRepositoryService repositoryService;
 
-  public AppEngineLibraryContainerResolverJob(String name, IJavaProject javaProject) {
-    this(name, javaProject, new LibraryClasspathContainerSerializer());
+  public AppEngineLibraryContainerResolverJob() {
+    this(new LibraryClasspathContainerSerializer());
```

Comment:
You could annotate `LibraryClasspathContainerSerializer` with [`@Creatable`](https://wiki.eclipse.org/Eclipse4/RCP/Dependency_Injection#.40Creatable_.28org.eclipse.e4.core.di.annotations.29) and then have it be `@Inject`ed.